service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java

```diff
@@ -423,7 +423,9 @@ private void searchAttributesWithScopes(final E entity, final Vertex vertex, fin
 
     private void checkTraversalLimitOrFail(final Set<Attribute> attributes) {
         if (attributes.size() > this.traversalLimit) {
-            throw new QueryException("Graph search failed: traversal limit exceeded.");
+            String exceptionMessage = String.format("Graph search failed: traversal limit of %s exceeded.",
+                    this.traversalLimit);
```

Comment:
It would be better to return the exact message that will be propagated upstream: "The number of attributes on this resource/subject have exceeded the maximum limit of %s."  This way later on you don't have to manipulate or construct another redundant message as written [here](https://github.com/predix/acs/blob/2c0d68993678945a4e5e62b808734f849e1c0055/service/src/main/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceImpl.java#L207-L228).